### DISCUSSION TITLE
[Workaround] Neovim syntax highlighting hangs when staging hunks

### DISCUSSION THREAD

**Discussion by AlxHnr** [tpope/vim-fugitive#2337] [General]
# Background

This issue has been happening for around a year now, but due to a lack of time I have not investigated whether it is my setup or something else. It is reproducible with Neovim on Alpine Linux and on macOS (installed via brew).

# Reproducing

1. Open a project with many changed hunks in a single file. When unfolding the diff in the `:Git` status window, the buffer should have around 1000 lines of highlighted diff
2. Stage/unstage the first hunk under the cursor with `-`
3. Neovim hangs for at least half a minute, sometimes even more

# Workaround

Set some lighter syntax highlighting:

```lua
vim.api.nvim_create_autocmd('User', {
  pattern = 'FugitiveIndex', group = YOUR_AUGROUP_HERE,
  callback = function() vim.bo.syntax = 'diff' end
})
```

**Comment by b0ae989c** [comment]
Could you turn off treesitter highlighting and try repro again?

**Reply by AlxHnr** [reply]
> Could you turn off treesitter highlighting and try repro again?

Still occurs, even with no treesitter highlighting.
